tests/numpy/test_creation_manipulation.py

```diff
@@ -78,6 +79,37 @@ def test_diag(nps_app_inst):
     np_arr = np.diag(np_arr)
     assert np.allclose(ba.get(), np_arr)
 
+def test_trace(nps_app_inst):
```

Comment:
Can you add a test for dtype?

Reply:
I have added a test for this, wherein the elements are like 1.01, 2.02, 3.03 and 4.04 and I need an int sum.